lib/models/ecs-model-deployer.ts

```diff
@@ -0,0 +1,68 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { Construct } from 'constructs';
+import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { Role, ServicePrincipal, ManagedPolicy } from 'aws-cdk-lib/aws-iam';
+import { Stack, Duration, Size } from 'aws-cdk-lib';
+
+import { createCdkId } from '../core/utils';
+import { BaseProps, Config } from '../schema';
+
+export type ECSModelDeployerProps = {
+    vpcId: string;
+    securityGroupId: string;
+    config: Config;
+} & BaseProps;
+
+export class ECSModelDeployer extends Construct {
+    constructor (scope: Construct, id: string, props: ECSModelDeployerProps) {
+        super(scope, id);
+        const stackName = Stack.of(scope).stackName;
+        const role = new Role(this, createCdkId([stackName, 'ecs-model-deployer-role']), {
+            assumedBy: new ServicePrincipal('lambda.amazonaws.com')
+        });
+
+        role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AdministratorAccess'));
```

Comment:
this is probably going to get flagged for being overly permissive. since its cloudformation and we need a ton of permissions anyways, could we at least restrict to the "*" policies for specific services we use? like `ec2:*`, `ecs:*`, `lambda:*`, and so on?